### PR TITLE
Fixes #18347: Group owner of files under configuration-repository are inconsistent

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/18347

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala [hunk 1]
```diff
@@ -724,7 +724,8 @@ class TechniqueArchiverImpl (
   , override val xmlPrettyPrinter          : RudderPrettyPrinter
   , override val relativePath              : String
   , override val gitModificationRepository : GitModificationRepository
-  , personIdentservice : PersonIdentService
+  , personIdentservice                     : PersonIdentService
+  , override val groupOwner                : String
```
Comment (ElaadF) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala:728]: I didn't know that we can override a method with members of the class

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/xml/GitArchiverUtils.scala [hunk 2]
```diff
@@ -200,14 +206,29 @@ trait GitArchiverUtils {
...
+    import better.files._
+    val filePerms = Set[PosixFilePermission](OWNER_READ, OWNER_WRITE, /* no exec, */ GROUP_READ, GROUP_WRITE /* no exec, */ )
+    val directoryPerms = Set[PosixFilePermission](OWNER_READ, OWNER_WRITE, OWNER_EXECUTE, GROUP_READ, GROUP_WRITE, GROUP_EXECUTE)
+
+    // an utility that write text in a file and create file parents if needed
+    // open file mode for create or overwrite mode
     IOResult.effect {
-      FileUtils.writeStringToFile(fileName, xmlPrettyPrinter.format(elem), encoding)
+      val file = fileName.toScala
+      file.parent.createDirectoryIfNotExists(true).setPermissions(directoryPerms).setGroup(groupOwner)
```
Comment (ElaadF) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/xml/GitArchiverUtils.scala:230]: Does the directory created is the same as `fileName` ?

Comment (fanf) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/xml/GitArchiverUtils.scala:230]: I don't understand the question. Can you explain a bit more?

**Review by ElaadF — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**